src/server/pfs/server/api_server.go

```diff
@@ -336,7 +349,12 @@ func (a *apiServer) InspectBranchInTransaction(txnCtx *txncontext.TransactionCon
 func (a *apiServer) ListBranch(request *pfs.ListBranchRequest, srv pfs.API_ListBranchServer) (retErr error) {
 	func() { a.Log(request, nil, nil, 0) }()
 	defer func(start time.Time) { a.Log(request, nil, retErr, time.Since(start)) }(time.Now())
-	return a.driver.listBranch(srv.Context(), request.Repo, request.Reverse, srv.Send)
+	if request.Repo == nil {
+		return a.driver.listBranch(srv.Context(), request.Reverse, srv.Send)
+	}
+	return a.txnEnv.WithReadContext(srv.Context(), func(txnCtx *txncontext.TransactionContext) error {
+		return a.driver.listBranchInTransaction(txnCtx, request.Repo, request.Reverse, srv.Send)
+	})
```

Comment:
I separated these two explicitly based on transaction compatibility, but I am happier to call out that this is a different path with potentially different auth requirements